File: code/Relative_complement_of_A_and_B.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class Relative_complement_of_A_and_B_2 {
    public static void main(String[] args) {
        List<String> A = new ArrayList<>(Arrays.asList("a","b","c","d","e"));
        List<String> B = new ArrayList<>(Arrays.asList("f","e","h","b","e"));

        System.out.println("Liste A: " + A);
        System.out.println("Liste B: " + B);

        A.removeAll(B); // Removes all elements of B that are also contained in A.
        System.out.println("------------------");
        System.out.println("Relative complement of A and B: " + A);
    }
}
// Ausgabe:
// Liste A: [a, b, c, d, e]
// Liste B: [f, e, h, b, e]
// ------------------
// Relative complement of A and B: [a, c, d]

File: code/Relative_complement_of_A_and_B.kt
fun main() {
    val a = mutableListOf("a", "b", "c", "d", "e")
    val b = mutableListOf("f", "e", "h", "b", "e")

    println("Liste A: $a")
    println("Liste B: $b")

    a.removeAll(b) // Removes all elements of b that are also contained in a.

    println("------------------")
    println("Relative complement of a and b: $a")

    // To determine the relative complement in Koltin, you can also use the minus operator:
    val relativeComp = a-b
    println("Relative complement of a and b: $relativeComp")
}
// Liste A: [a, b, c, d, e]
// Liste B: [f, e, h, b, e]
// ------------------
// Relative complement of a and b: [a, c, d]
// Relative complement of a and b: [a, c, d]